### ISSUE TITLE
--expand not properly documented.

### ISSUE THREAD

**Issue by SamMousa** [certbot/certbot#3664]
From the docs:

```
  --expand              If an existing cert covers some subset of the
                        requested names, always expand and replace it with the
                        additional names. (default: False)
```

Existing certificate has: `example.com` and `www.example.com`.
Works:
`letsencrypt-auto --expand --force-renewal certonly -d example.com -d test.example.com -d www.example.com`

Doesn't work:
`letsencrypt-auto --expand certonly -d example.com -d test.example.com`

Note that there are 2 differences between these commands:
1. `--force-renewal`, I replicated what I used to get it to work; but this is a prod server so I'm hesitant to retry and risk getting rate limited.
1. We are not listing all existing domains on the existing certificate.

The issue is that 2 should not be a problem according to docs. Even more so, expanding should always expand if any certificate exists. "If an existing certificate covers some subset of the requested names...", the empty subset is always covered and requested. Also what if an existing certificate covers a superset of the requested names ("covers" to me suggests that would be fine)? It should then expand and add the newly requested domains.

In conclusion:
1. Does --expand work without --force-renewal?
2. Can I match an existing certificate by just specifying a subset of covered domains?

**Comment by bmw** [comment]
Yes you can use `--expand` without `--force-renewal`. The problem is that you're not listing all domains in the existing certificate. Perhaps we should change the documentation for this flag to say "If an existing cert **is a** subset of the requested names...".

> Also what if an existing certificate covers a superset of the requested names ("covers" to me suggests that would be fine)? It should then expand and add the newly requested domains.

In this case, Certbot would create a new directory in `/etc/letsencrypt/live` for the certificate. I believe this problem is also resolved if we change "covers" to "is a".

Honestly, the `--expand` flag is a bit clunky and supporting use cases like this should be a lot easier in our next release. See #3615 for what we're planning to do.

**Comment by SwartzCr** [comment]
tagging @pconrad-fb who might want to add documentation for this in his lineage docuementation

**Comment by SwartzCr** [comment]
actually, since #3615 has landed, where does the state of this issue stand, @bmw is it closable?

**Comment by bmw** [comment]
No this isn't closable, however, it's just about changing the help for `--expand` in Certbot.

**Comment by SwartzCr** [comment]
got it - opened a PR

**Comment by SwartzCr** [comment]
it's #4495

**Comment by bmw** [comment]
Fixed in #4495.

**Comment by pconrad-fb** [comment]
Looking at the doc, I see the following for --expand:

> ``--expand`` tells Certbot to update an existing certificate with a new
> certificate that contains all of the old domains and one or more additional
> new domains.

Should I add the following sentence: "With the ``--expand`` option, you must list all the domains currently in the certificate as well as the domains you are adding."

?

**Comment by SwartzCr** [comment]
@bmw thoughts?

**Comment by bmw** [comment]
Yep. SGTM!

**Comment by SwartzCr** [comment]
@pconrad-fb if you want to make a PR with that change I'd be happy to review and merge it :D

**Comment by pconrad-fb** [comment]
Sorry, forgot to @mention for visibility: @schoen @SwartzCr see above comment for a syntax question.

**Comment by bmw** [comment]
> certbot -renew --expand -d existing.com,example.com,newdomain.com
>
> certbot -renew --expand -d existing.com -d example.com -d newdomain.com

If you run either of these commands, you get:
```
certbot: error: unrecognized arguments: -renew
```
> certbot --expand -d existing.com,example.com,newdomain.com
>
> certbot --expand -d existing.com -d example.com -d newdomain.com

These commands are equivalent though.
